File: Week_01/move_zeroes/Solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        first_zero_index = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[i], nums[first_zero_index] = nums[first_zero_index], nums[i]
                first_zero_index += 1

File: Week_01/move_zeroes/Solution.java
package move_zeroes;

//https://leetcode-cn.com/problems/move-zeroes

/**
 * 双指针
 * 时间复杂度：O(n)
 * 空间复杂度：O(1)
 */
class Solution {
    public void moveZeroes(int[] nums) {
        for (int i = 0, firstZeroAt = 0; i < nums.length; i++)
            if (nums[i] != 0) {
                int temp = nums[i];
                nums[i] = nums[firstZeroAt];
                nums[firstZeroAt++] = temp;
            }
    }
}
